libfreerdp/core/timezone.c

```diff
@@ -117,53 +113,37 @@ BOOL rdp_read_client_time_zone(wStream* s, rdpSettings* settings)
 
 void rdp_write_client_time_zone(wStream* s, rdpSettings* settings)
 {
-	WCHAR* standardName = NULL;
-	WCHAR* daylightName = NULL;
-	int standardNameLength;
-	int daylightNameLength;
-	TIME_ZONE_INFO* clientTimeZone;
-
-	clientTimeZone = settings->ClientTimeZone;
-	freerdp_time_zone_detect(clientTimeZone);
-
-	standardNameLength = ConvertToUnicode(CP_UTF8, 0, clientTimeZone->standardName, -1, &standardName, 0) * 2;
-	daylightNameLength = ConvertToUnicode(CP_UTF8, 0, clientTimeZone->daylightName, -1, &daylightName, 0) * 2;
+	LPTIME_ZONE_INFORMATION tz;
+	DWORD rc;
 
-	if (standardNameLength > 62)
-		standardNameLength = 62;
+	tz = settings->ClientTimeZone;
+	rc = GetTimeZoneInformation(tz);
```

Comment:
I might be mistaken but I think GetTImeZoneInformation isn't implemented (as in return 0).